server/src/controllers/appointments.controller.js

```diff
@@ -1,6 +1,27 @@
 import Appointment from "../models/appointments.model.js";
 import appointmentSchema from "../validators/appointments.validator.js";
 
+export async function geocodeAddress(address) {
+  const geocodeUrl = `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${process.env.GOOGLE_API_KEY}`;
+
+  try {
+    const resp = await fetch(geocodeUrl);
+    const data = await resp.json();
+    console.log(data);
+    if (data.status === "OK") {
+      const geodata = data.results[0].geometry.location;
+      console.log("geodata", geodata);
+
+      return geodata;
+    } else {
+      throw new Error(`Geocoding issue: ${data.status}`);
+    }
```

Comment:
Quite weird to throw an error then immediately re-catch it. Would suggest if you want to throw something like this, it should be outside the try catch